### PR TITLE
[plugin-mobile-app] Add step to type keys

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 1]
```diff
@@ -658,4 +658,24 @@ When I press keys:
 |$  |
 ----
 
+=== Type keys
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:661]:
```suggestion
=== Type text
```

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 2]
```diff
@@ -658,4 +658,24 @@ When I press keys:
...
+
+Types a sequence of keys
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:663]:
```suggestion
Types the provided text
```

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 3]
```diff
@@ -658,4 +658,24 @@ When I press keys:
...
+
+[WARNING]
+All the characters will be treated as is and passed sequentially. To send special key like `HOME` please use <<_press_keys>>
+
+[source,gherkin]
+----
+When I type keys `$keys`
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:670]:
```suggestion
When I type text `$text`
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/DeviceSteps.java [hunk 4]
```diff
@@ -118,6 +136,23 @@ public void pressKeys(ExamplesTable keys)
         pressKeys(keys.getColumn("key"));
     }
 
+    /**
+     * Types the text into the focused field
+     * @param text the text to type
+     */
+    @When("I type text `$text`")
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/DeviceSteps.java:143]: https://github.com/vividus-framework/vividus/blob/master/vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java#L104 -  I guess we need to have similar logic in both steps

**Review by uarlouski — APPROVED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java [hunk 5]
```diff
@@ -152,6 +185,91 @@ public void swipeToElement(SwipeDirection direction, Locator locator, Duration s
         }
     }
 
+
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:188]: extra line

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java [hunk 6]
```diff
@@ -16,43 +16,76 @@
...
+            ISearchActions searchActions, GenericWebDriverManager genericWebDriverManager,
+            JavascriptActions javascriptActions, IWebDriverProvider webDriverProvider)
     {
         this.touchActions = touchActions;
         this.keyboardActions = keyboardActions;
         this.baseValidations = baseValidations;
         this.searchActions = searchActions;
         this.genericWebDriverManager = genericWebDriverManager;
+        this.javascriptActions = javascriptActions;
+        this.webDriverProvider = webDriverProvider;
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:88]: it looks like this class is overloaded, does it make sense to create a new class, e.g. `KeyboardSteps`?

Comment (ikalinin1) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:88]: image

File: vividus-plugin-mobile-app/src/test/java/org/vividus/bdd/mobileapp/steps/TouchStepsTests.java [hunk 7]
```diff
@@ -47,24 +47,26 @@
...
 import org.vividus.ui.action.search.SearchParameters;
 
 @ExtendWith(MockitoExtension.class)
 class TouchStepsTests
 {
-    private static final String TEXT = "text";
     private static final String ELEMENT_TO_TAP = "The element to tap";
-    private static final String ELEMENT_TO_TYPE_TEXT = "The element to type text";
 
     @Mock private IBaseValidations baseValidations;
```
Comment (uarlouski) [inline vividus-plugin-mobile-app/src/test/java/org/vividus/bdd/mobileapp/steps/TouchStepsTests.java:62]: are all of these mocks required?

**Review by valfirst — APPROVED** [review]

**Pull request merged**